packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,541 @@
+
+#### Heavy sized library - AG Grid
+- [AG Grid](https://www.ag-grid.com/) is the best example of a heavy sized library. It has a lot of features and is very well maintained. However, it is also very heavy and has a lot of features that we don't need.
+- As of today, AG Grid's bundle size is 75kb (15kb gzipped) but along with this we also need to install some additional dependency libraries like ag-grid-community which would eventually increase the overall bundle size
+- Replicating the Composable API would be a challenging task with React Table since it internally uses API similar to the Compact API. We would have to build a lot of abstractions to ensure we can build the Composable API on top of the Compact API.
+- AG Grid is possibly the most battle-tested library but has a ton of features that we won't need and hence would be an overkill for us as of now
+
+#### Medium sized library - Tanstack's React Table
+- [Tanstack's React Table](https://react-table.tanstack.com/) is the best example of a medium sized library. It has all the features we need and is very well maintained. However, it is also relatively heavy and has a lot of features that we don't need.
+- As of today, React Table's bundle size is 60kb (14kb gzipped).
+- Tanstack's React  Table has been around for a longer duration and is extremely well maintained. It has relatively less features than AG Grid but also has a lot of features that we don't need.
+- Tanstack's React Table is built in a framework agnostic manner which meant they had to use a lot of abstractions to ensure it works well with all the frameworks. This makes it harder to understand, maintain and extend the library.
+- Replicating the Composable API would be a challenging task with React Table since it internally uses API similar to the Compact API. We would have to build a lot of abstractions to ensure we can build the Composable API on top of the Compact API.
+- React Table is possibly our 2nd best option if we ever need more complex tables in the future but as of now it would be an overkill for us
+
+#### Small sized library - React Table Library
+- [React Table Library (@table-library/react-table-library)](https://github.com/table-library/react-table-library) is the best example of a small sized library. It has all the features we need and is relatively well maintained.
+- It contains all the features we need today as well as the ones we can forsee needing in the future
+- As of today, React Table Library's bundle size is 23kb (7kb gzipped).
+- React Table Library has a Composable API using which we can build our Composable as well as Compact API
+- React Table Library also exposes a minimal Compact API but we cannot use that directly to build our Compact API since there isn't a parity between their own Composable & Compact APIs
+- React Table Library is our best option as of today since it has all the features we could need and is relatively lightweight
+
```

Comment:
We should perhaps also consider built-in virtualization for tables, otherwise consumers may have to do the wiring themselves 🤔

Reply:
As long as we use proper pagination, virtualization might not be needed at the moment. But there is support for virtualization for all the table libraries we have evaluated and selected.